grakn-graql/src/test/java/ai/grakn/graql/internal/analytics/CorenessTest.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql.internal.analytics;
+
+import ai.grakn.GraknSession;
+import ai.grakn.GraknTx;
+import ai.grakn.GraknTxType;
+import ai.grakn.concept.Attribute;
+import ai.grakn.concept.AttributeType;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.concept.Entity;
+import ai.grakn.concept.EntityType;
+import ai.grakn.concept.RelationshipType;
+import ai.grakn.concept.Role;
+import ai.grakn.exception.GraqlQueryException;
+import ai.grakn.exception.InvalidKBException;
+import ai.grakn.graql.Graql;
+import ai.grakn.test.rule.SessionContext;
+import ai.grakn.util.GraknTestUtil;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeFalse;
+
+public class CorenessTest {
+    private static final String thing = "thingy";
+    private static final String anotherThing = "anotherThing";
+    private static final String related = "related";
+    private static final String veryRelated = "veryRelated";
+
+    private ConceptId entityId1;
+    private ConceptId entityId2;
+    private ConceptId entityId3;
+    private ConceptId entityId4;
+
+    public GraknSession session;
+
+    @ClassRule
+    public final static SessionContext sessionContext = SessionContext.create();
+
+    @Before
+    public void setUp() {
+        session = sessionContext.newSession();
+    }
+
+    @Test(expected = GraqlQueryException.class)
+    public void testKSmallerThan2() {
+        try (GraknTx graph = session.open(GraknTxType.READ)) {
+            graph.graql().compute().coreness().minK(1).execute();
+        }
+    }
+
+    @Test
+    public void testOnEmptyGraph() {
```

Comment:
We can also say the expected result here, like `testOnEmptyGraph_ReturnsEmptyMap`. Same for the tests below too!